src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4558,6 +4585,8 @@ int wm_vuldet_fetch_feed(update_node *update, int8_t *need_update) {
 
 int wm_vuldet_check_feed(update_node *upd, int8_t *updated) {
     if (wm_vuldet_check_update_period(upd)) {
+        vu_task_ctx_t *task = create_task("%lu %d %d %d", 1, VU_INTERVAL_FEED_UPDATE, WM_TASK_PENDING, upd->dist_ref);
```

Comment:
The methods still are heavily coupled so it isn't possible to determine if it's time to update the feed without actually updating it.

We should discuss with the team the scope of the issue.

Reply:
I made some improvements, in function `wm_vuldet_run_update()` after enqueue the task we perform the feed update but it could easily be moved to another method to be executed for the working thread.